src/report/reportUtils.ts

```diff
@@ -1,49 +1,53 @@
-import { Inspection } from '../inspection/InspectionEntity'
+import type { Inspection } from '../inspection/InspectionEntity'
 import { BaseReport } from './ReportEntity'
 import { WorkBook } from 'xlsx'
 import { omit } from 'lodash'
 import { Constructor } from '../common/type/common'
 import { cacheFetch, clearCacheKey } from '../cache'
 import { BaseReportData } from './BaseReportDataEntities'
 import { getRepository } from 'typeorm'
-import { IReportArgs, IReportConfig } from './reportTypes'
+import { IReportArgs, IReportConfig, ReportInspectionType } from './reportTypes'
 import { createPagedResponseData } from '../util/filteredResponseUtils'
-import { ApolloError } from 'apollo-server-express'
 import { reportNames } from './reportNames'
 import { getInspectionStatusValue } from '../inspection/inspectionStates'
+import { PreInspection } from '../inspection/PreInspectionEntity'
+import { PostInspection } from '../inspection/PostInspectionEntity'
+import { getInspectionById, isPreInspection } from '../inspection/inspectionUtils'
 
-export function createReportCacheKey(inspection: Inspection, reportName) {
+export function createReportCacheKey<InspectionType extends Inspection>(
```

Comment:
Is
```
InspectionType extends Inspection
```
Needed?

```suggestion
export function createReportCacheKey(
  inspection: Inspection,
  reportName
) {
```
Can it be done just like this? I mean that generic type declaration <...> is not used at least here

Reply:
It's a type constraint. If it would be just `Inspection` you couldn't pass Pre or PostInspections to this function. It is "used" just because it exists.

Reply:
The idea is that it is a type constraint, but now that I tested, just `inspection: Inspection` without the generic thing seems to work just as well! I was under the impression that this needed to be wired up with generics.